src/transformers/models/vilt/image_processing_vilt_fast.py

```diff
@@ -0,0 +1,264 @@
+
+        new_heights = int(new_heights + 0.5)
+        new_widths = int(new_widths + 0.5)
+
+        # Make dimensions divisible by size_divisor
+        if size_divisor is not None:
+            new_heights = new_heights // size_divisor * size_divisor
+            new_widths = new_widths // size_divisor * size_divisor
+
+        # Resize the image
+        return F.resize(images, [new_heights, new_widths], interpolation=interpolation)
+
+    def _pad_batch(
+        self,
+        images: list["torch.Tensor"],
+        return_tensors: Optional[Union[str, TensorType]],
+    ) -> tuple:
+        """
+        Pad a batch of images to the same size based on the maximum dimensions.
+
+        Args:
+            images (`list[torch.Tensor]`): List of images to pad.
+            return_tensors (`str` or `TensorType`, *optional*): The type of tensors to return.
+
+        Returns:
+            `tuple`: Tuple containing padded images and pixel masks.
+        """
+        # Calculate global maximum dimensions across all images
+        max_size = get_max_height_width(images)
+
+        # Group images by shape before padding
+        grouped_images, grouped_images_index = group_images_by_shape(images)
+        processed_grouped = {}
+
+        for shape, stacked_images in grouped_images.items():
+            # Create mask template for efficient masking
+            if return_tensors == "pt" and len(stacked_images) > 0:
+                device = stacked_images.device
+                mask_template = torch.zeros(max_size, dtype=torch.int64, device=device)
+
+            original_size = stacked_images.shape[-2:]
+            needs_padding = original_size[0] != max_size[0] or original_size[1] != max_size[1]
+
+            if needs_padding:
+                padding_bottom = max_size[0] - original_size[0]
+                padding_right = max_size[1] - original_size[1]
+                padding = [0, 0, padding_right, padding_bottom]
+
+                padded_images = F.pad(stacked_images, padding, fill=0)
+                pixel_mask = mask_template.clone()
+                pixel_mask[: original_size[0], : original_size[1]].fill_(1)
+                pixel_masks = pixel_mask.unsqueeze(0).repeat(stacked_images.shape[0], 1, 1)
+            else:
+                padded_images = stacked_images
+                pixel_masks = torch.ones(
+                    (stacked_images.shape[0], max_size[0], max_size[1]),
+                    dtype=torch.int64,
+                    device=stacked_images.device,
+                )
+
+            # Store processed group
+            processed_grouped[shape] = (padded_images, pixel_masks)
+
+        # Reorder images back to original order
+        padded_images = []
+        pixel_masks = []
+
+        for _, (group_key, position) in grouped_images_index.items():
+            padded_images.append(processed_grouped[group_key][0][position])
+            pixel_masks.append(processed_grouped[group_key][1][position])
```

Comment:
```suggestion
        padded_images= reorder_images(processed_images, grouped_images_index)
        pixel_masks= reorder_images(processed_masks, grouped_images_index)
```
Let me know if something like this works!